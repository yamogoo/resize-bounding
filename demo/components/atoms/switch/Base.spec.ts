import { describe, expect, test } from "vitest";
import { mount } from "@vue/test-utils";

import Base from "./Base.vue";

describe("Switch", () => {
  test.each([true])(
    'should add class attribute "active" to container DOM if "state" prop equals %s',
    (state) => {
      const wrapper = mount(Base, {
        props: {
          state,
        },
      });

      const className = wrapper.classes("active");
      expect(className).toBe(true);
      expect(wrapper.classes("normal")).toBe(false);
      expect(className).toMatchSnapshot();
    },
  );

  test.each([false])(
    'should add class attribute "normal" to container DOM if "state" prop equals %s',
    (state) => {
      const wrapper = mount(Base, {
        props: {
          state,
        },
      });

      const className = wrapper.classes("normal");
      expect(className).toBe(true);
      expect(wrapper.classes("active")).toBe(false);
      expect(className).toMatchSnapshot();
    },
  );

  test('should emit "update:state" when clicking on the wrapper', async () => {
    const wrapper = mount(Base, {
      props: {
        state: true,
      },
    });

    const inputEl = wrapper.find('input[type="checkbox"]');
    await inputEl.trigger("change");

    expect(wrapper.emitted("update:state")).toHaveLength(1);
    expect(wrapper.emitted()).toHaveProperty("update:state");
    expect(wrapper.emitted("update:state")).toMatchSnapshot();
  });
});
