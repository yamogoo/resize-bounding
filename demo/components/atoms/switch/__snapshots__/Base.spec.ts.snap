// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Switch > should add class attribute "active" to container DOM if "state" prop equals true 1`] = `true`;

exports[`Switch > should add class attribute "normal" to container DOM if "state" prop equals false 1`] = `true`;

exports[`Switch > should emit "update:state" when clicking on the wrapper 1`] = `
[
  [
    true,
  ],
]
`;
